hack/install.sh

```diff
@@ -231,10 +231,60 @@ do_install() {
 			exit 0
 			;;
 
-		'opensuse project'|opensuse|'suse linux'|sle[sd])
+		'opensuse project'|opensuse)
+			echo 'Going to perform the following operations:'
+			if [ "$repo" != 'main' ]; then
+				echo '  * add repository obs://Virtualization:containers'
+			fi
+			echo '  * install Docker'
+			$sh_c 'echo "Press CTRL-C to abort"; sleep 3'
+
+			if [ "$repo" != 'main' ]; then
+				# install experimental packages from OBS://Virtualization:containers
+				(
+					set -x
+					zypper -n ar -f obs://Virtualization:containers Virtualization:containers
+					$sh_c 'curl -sSL https://api.opensuse.org/public/source/Virtualization/_pubkey > /tmp/virtualization.key && rpm --import /tmp/virtualization.key && rm /tmp/virtualization.key'
+				)
+			fi
+			(
+				set -x
+				zypper -n install docker
+			)
+			echo_docker_as_nonroot
+			exit 0
+			;;
+		'suse linux'|sle[sd])
+			echo 'Going to perform the following operations:'
+			if [ "$repo" != 'main' ]; then
+				echo '  * add repository obs://Virtualization:containers'
+				echo '  * install experimental Docker using packages NOT supported by SUSE'
+			else
+				echo '  * add the "Containers" module'
+				echo '  * install Docker using packages supported by SUSE'
+			fi
+			$sh_c 'echo "Press CTRL-C to abort"; sleep 3'
+
+			if [ "$repo" != 'main' ]; then
+				# install experimental packages from OBS://Virtualization:containers
+				echo >&2 'Warning: installing experimental packages from OBS, these packages are NOT supported by SUSE'
+				(
+					set -x
+					zypper -n ar -f obs://Virtualization:containers/SLE_12 Virtualization:containers
+					$sh_c 'curl -sSL https://api.opensuse.org/public/source/Virtualization/_pubkey > /tmp/virtualization.key && rpm --import /tmp/virtualization.key && rm /tmp/virtualization.key'
```

Comment:
The packages are built exactly on api.opensuse.org, they are later published to the url download.opensuse.org which is then mirrored to different locations. The client then picks up the closes mirror thanks to some files created by mirrorbrain.

So, checking the key provided by api.opensuse.org is the right thing to do.